### PR TITLE
feat: Spark 3.0

### PR DESCRIPTION
**Pull request by dciborow**
ready to be merged or rebased

Co-authored-by: Mark Hamilton <marhamil@microsoft.com>

### CODE REVIEW

File: src/main/scala/com/microsoft/ml/spark/featurize/AssembleFeatures.scala [hunk 1]
```diff
@@ -443,7 +443,7 @@ class AssembleFeaturesModel(val uid: String,
         val outputColsKeys = columnNamesToFeaturize.categoricalColumns.values
         val inputCols = inputColsKeys.toArray
         val outputCols = outputColsKeys.toArray
-        val ohe = new OneHotEncoderEstimator()
+        val ohe = new OneHotEncoder()
```
Comment (imatiach-msft) [inline src/main/scala/com/microsoft/ml/spark/featurize/AssembleFeatures.scala:446]: is there some way to make this be based on the currently used spark version?

**Review by imatiach-msft — DISMISSED** [review]

File: src/main/scala/com/microsoft/ml/spark/featurize/AssembleFeatures.scala (hunk 1 above)
Comment (dciborow) [reply src/main/scala/com/microsoft/ml/spark/featurize/AssembleFeatures.scala:446]: Can you do something like...
val ohe = if (SparkContext.version < 3.0) new OneHotEncoderEstimator() else new OneHotEncoder()

**Comment by Fokko** [comment]
Any progress on this? We're using Spark 3.0 and this package looks promising.

cc @jannickseelenah

**Comment by hanbing1587** [comment]
@imatiach-msft @dciborow @mhamilton723. Did you ever build this branch successfully? I try to build this branch but get 19 errors. Do you know how to solve this issue?

File: src/main/scala/com/microsoft/ml/spark/featurize/AssembleFeatures.scala (hunk 1 above)
Comment (blainedietrich) [reply src/main/scala/com/microsoft/ml/spark/featurize/AssembleFeatures.scala:446]: Yeah @dciborow sounds good, do that!

**Pull request closed without merging**